app/controllers/products_controller.rb

```diff
@@ -32,4 +32,17 @@ def update
       end
     end
   end
+
+  def vote
+    vote = Vote.where(:user_id => current_user.id, :product_id => params[:product_id]).first
+    if vote.present?
+      vote.destroy
+    else
+      Vote.create(:user_id => current_user.id, :product_id => params[:product_id])    
+    end
+  end
+
+  def vote_count
```

Comment:
the `vote_count` should be on the products JSON response.

No need for a separate endpoint.